### ISSUE TITLE
listen() returning error

### ISSUE THREAD

**Issue by stefanhdao** [carpedm20/fbchat#72]
Getting the following error from the listen() function.

```
Exception in thread Thread-2:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "fb_chat.py", line 59, in handle_new_message
    client.listen()
  File "/usr/local/lib/python2.7/dist-packages/fbchat/client.py", line 517, in listen
    sticky, pool = self._getSticky()
  File "/usr/local/lib/python2.7/dist-packages/fbchat/client.py", line 442, in _getSticky
    j = get_json(r.text)
  File "/usr/local/lib/python2.7/dist-packages/fbchat/utils.py", line 18, in get_json
    return json.loads(re.sub(r"^[^{]*", '', text, 1))
  File "/usr/lib/python2.7/json/__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python2.7/json/decoder.py", line 384, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded
```

With the following lines,

```
print r
print '-----'
print r.text
```

I got this output

```
<Response [400]>
-----

```

**Comment by Dgameman1** [comment]
I just get this error

```
Traceback (most recent call last):
  File "/opt/pycharm-com-eap/helpers/pydev/pydevd.py", line 1581, in <module>
    globals = debugger.run(setup['file'], None, None, is_module)
  File "/opt/pycharm-com-eap/helpers/pydev/pydevd.py", line 964, in run
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "/home/daniel/Dropbox/Code/Python/TesFBChat.py", line 19, in <module>
    bot.listen()
  File "/home/daniel/.local/lib/python2.7/site-packages/fbchat/client.py", line 544, in listen
    sticky, pool = self._getSticky()
  File "/home/daniel/.local/lib/python2.7/site-packages/fbchat/client.py", line 468, in _getSticky
    j = get_json(r.text)
  File "/home/daniel/.local/lib/python2.7/site-packages/fbchat/utils.py", line 18, in get_json
    return json.loads(re.sub(r"^[^{]*", '', text, 1))
  File "/usr/lib/python2.7/json/__init__.py", line 339, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 364, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python2.7/json/decoder.py", line 382, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded
```

**Comment by TriRpi** [comment]
I have the same issue

**Comment by dstelzer** [comment]
I'm getting a similar but slightly different error.

    >>> c = fbchat.Client('EMAIL', 'PASSWORD')
    Logging in...
    login successful
    >>> c.listen()
    Traceback (most recent call last):
      File "<stdin>", line 1, in <module>
      File "/usr/local/lib/python3.5/dist-packages/fbchat/client.py", line 544, in listen
        sticky, pool = self._getSticky()
      File "/usr/local/lib/python3.5/dist-packages/fbchat/client.py", line 468, in _getSticky
        j = get_json(r.text)
      File "/usr/local/lib/python3.5/dist-packages/fbchat/utils.py", line 18, in get_json
        return json.loads(re.sub(r"^[^{]*", '', text, 1))
      File "/usr/lib/python3.5/json/__init__.py", line 319, in loads
        return _default_decoder.decode(s)
      File "/usr/lib/python3.5/json/decoder.py", line 339, in decode
        obj, end = self.raw_decode(s, idx=_w(s, 0).end())
      File "/usr/lib/python3.5/json/decoder.py", line 357, in raw_decode
        raise JSONDecodeError("Expecting value", s, err.value) from None
    json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

**Comment by Zulex** [comment]
Getting the same error here too, worked a while ago, didn't change anything.

**Comment by Zulex** [comment]
Fix is written here : https://github.com/carpedm20/fbchat/pull/77

**Comment by vlegoff** [comment]
Thanks @Zulex, it seems to fix the bug.  I hope it will be merged soon.

**Comment by tshrjn** [comment]
Getting the same error as @dstelzer , and @Zulex , the fix you specified does not help either.

**Comment by vlegoff** [comment]
I tried the fix, and it seemed to work all right.  You just need to use 
the fork... and update to the branch where the fix is (the branch master 
is identical to the original branch).  I hope this fix gets accepted!


HTH,

**Comment by carpedm20** [comment]
Merged #77
